06x-new-kafka-roller.md

```diff
@@ -0,0 +1,350 @@
+
+#### NOT_RUNNING state
+
+If one of the following is true, then node's state is `NOT_RUNNING:
+- no pod exists for this node
+- unable to get the `Pod Status` for the pod 
+- the pod has `Pending` status with `Unschedulable` reason
+- the pod has container status `ContainerStateWaiting` with `CrashLoopBackOff` or `ImagePullBackOff` reason
+If none of the above is true but the node is not ready, then its state would be `NOT_READY`.
+
+#### Flow diagram describing the overall flow of the states
+![The new roller flow](./images/06x-new-roller-flow.png)
+
+
+### Configurability
+The following can be the configuration options for the new KafkaRoller:
+
+| Configuration          | Default value | Exposed to user | Description                                                                                                                                                                                                                                                   |
+|:-----------------------|:--------------|:----------------|:--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| maxRestartAttempts     | 3             | No              | The maximum number of times a node can be restarted before failing the reconciliation. This is checked against the node's `numRestartAttempts`.                                                                                                               |
+| maxReconfigAttempts    | 3             | No              | The maximum number of times a node can be reconfigured before restarting it. This is checked against the node's `numReconfigAttempts`.                                                                                                                        |
+| maxRetries             | 10            | No              | The maximum number of times a node can be retried after not meeting the safety conditions. This is checked against the node's `numRetries`.                                                                                                                      |
+| postOperationTimeoutMs | 60 seconds    | Yes             | The maximum amount of time we will wait for nodes to transition to `SERVING` state after an operation in each retry. This will be based on the operation timeout that is already exposed to the user via environment variable `STRIMZI_OPERATION_TIMEOUT_MS`. |
+| maxRestartParallelism  | 1             | Yes             | The maximum number of broker nodes that can be restarted in parallel.                                                                                                                                                                                         |
+
+### Algorithm
```

Comment:
I found the following section difficult to read. Maybe we should try to structure it to make it easier.
By "structure" I mean not having just a bunch of sentences saying "if this then this but if this then that" and so on. But more something like:

<STATE> - <INPUT> - <OUTPUT> - <NEXTSTATE>

Where <INPUT> would be the observation you make and what you have from the context I assume. The <OUTPUT> would be any action taken on nodes (and maybe any update on the context? maybe not because it's rebuilt on each reconciliation and not stored anywhere).
I am just trying to figure out a way to make the algorithm more understandable. This comment copes with the one on the FSM diagram as well.

Reply:
I will take this into consideration and see how we can present the algorithm better without too much repetition which was the aim with it.